tests/test_on_data/test_gin_ecephys.py

```diff
@@ -235,34 +235,50 @@ class TestConverter(NWBConverter):
     )
     def test_neuroscope_gains(self, name, conversion_options):
         input_gain = 2.0
-        data_interface = NeuroscopeRecordingInterface
         interface_kwargs = dict(file_path=str(DATA_PATH / "neuroscope" / "test1" / "test1.dat"), gain=input_gain)
 
-        nwbfile_path = str(self.savedir / f"{data_interface.__name__}-{name}.nwb")
+        nwbfile_path = str(self.savedir / f"test_neuroscope_gains_{name}.nwb")
 
         class TestConverter(NWBConverter):
-            data_interface_classes = dict(TestRecording=data_interface)
+            data_interface_classes = dict(TestRecording=NeuroscopeRecordingInterface)
 
         converter = TestConverter(source_data=dict(TestRecording=interface_kwargs))
         converter.run_conversion(nwbfile_path=nwbfile_path, overwrite=True, conversion_options=conversion_options)
 
-        # nwb file check-test
-        io = NWBHDF5IO(nwbfile_path, "r")
-        nwbfile_in = io.read()
-        output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
-        output_gain = output_conversion * 1e6
-        assert input_gain == pytest.approx(output_gain)
+        with NWBHDF5IO(path=nwbfile_path, mode="r") as io:
+            nwbfile_in = io.read()
+            output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
+            output_gain = output_conversion * 1e6
+            self.assertEqual(first=input_gain, second=output_gain)
```

Comment:
@h-mayorquin It's good practice to use context managers whenever possible when dealing with file I/O, to ensure safe opening/closing/access